eng/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -154,45 +185,83 @@ function ProcessExcludes([Array]$matrix, [Array]$excludes)
     return $matrix | Where-Object { !$_.parameters.Contains($deleteKey) }
 }
 
-function ProcessIncludes([Array]$matrix, [Array]$includes, [Hashtable]$displayNamesLookup)
+function ProcessIncludes([MatrixConfig]$config, [Array]$matrix)
 {
-    foreach ($inclusion in $includes) {
-        $converted = ConvertToMatrixArrayFormat $inclusion
-        $full = GenerateFullMatrix $converted $displayNamesLookup
-        $matrix += $full
+    $inclusionMatrix = @()
+    foreach ($inclusion in $config.include) {
+        $full = GenerateFullMatrix $inclusion $config.displayNamesLookup
+        $inclusionMatrix += $full
     }
 
-    return $matrix
+    return $matrix + $inclusionMatrix
 }
 
-function MatrixElementMatch([System.Collections.Specialized.OrderedDictionary]$source, [System.Collections.Specialized.OrderedDictionary]$target)
+function ProcessImport([System.Collections.Specialized.OrderedDictionary]$matrix, [String]$selection)
 {
-    if ($target.Count -eq 0) {
-        return $false
+    if (-not $matrix -or -not $matrix.Contains($IMPORT_KEYWORD)) {
+        return $matrix
     }
 
-    foreach ($key in $target.Keys) {
-        if (-not $source.Contains($key) -or $source[$key] -ne $target[$key]) {
-            return $false
+    $importPath = $matrix[$IMPORT_KEYWORD]
+    $matrix.Remove($IMPORT_KEYWORD)
+
+    $matrixConfig = GetMatrixConfigFromJson (Get-Content $importPath)
+    $importedMatrix = GenerateMatrix $matrixConfig $selection
+
+    return $matrix, $importedMatrix
+}
+
+function CombineMatrices([Array]$matrix1, [Array]$matrix2)
+{
+    $combined = @()
+    if (-not $matrix1) {
+        return $matrix2
+    }
+    if (-not $matrix2) {
```

Comment:
nit: I generally prefer using the less verbose `!` operator over `-not`, they are the same but `!` feels more natural to me.